journal/src/main/scala/com/evolutiongaming/kafka/journal/util/PlayJsonHelper.scala

```diff
@@ -75,17 +81,20 @@ object PlayJsonHelper {
   }
 
 
-  def encodeByNameFromWrites[F[_], A](implicit writes: Writes[A], toTry: ToTry[F], encode: JsonCodec.Encode[F]): EncodeByName[A] =
+  def encodeByNameFromWrites[A](implicit writes: Writes[A], encode: JsonCodec.Encode[Try]): EncodeByName[A] = {
     EncodeByName[String].contramap { a =>
       val jsValue = writes.writes(a)
-      toTry(encode.toStr(jsValue)).get
+      encode.toStr(jsValue).get
     }
+  }
 
   
-  def decodeByNameFromReads[A](implicit reads: Reads[A]): DecodeByName[A] = DecodeByName[String].map { a =>
-    val jsValue = PlayJsonJsoniter.deserializeFromStr(a) getOrElse Json.parse(a)
-    // TODO not use `as`
-    jsValue.as(reads)
+  def decodeByNameFromReads[A](implicit reads: Reads[A], decode: JsonCodec.Decode[Try]): DecodeByName[A] = {
+    DecodeByName[String].map { str =>
+      val jsValue = decode.fromStr(str).get
```

Comment:
although I can accept the review, could you please benchmark this against direct PlayJson from String ?